rest_framework_gis/schema.py

```diff
@@ -0,0 +1,168 @@
+import warnings
+
+from django.contrib.gis.db import models
+from rest_framework.schemas.openapi import AutoSchema
+from rest_framework.utils import model_meta
+
+from rest_framework_gis.fields import GeometrySerializerMethodField
+from rest_framework_gis.serializers import GeoFeatureModelListSerializer, GeoFeatureModelSerializer
+
+
+class GeoFeatureAutoSchema(AutoSchema):
+    COORDINATES_SCHEMA_FOR_POINT = {
+        'type': 'array',
+        'items': {
+            'type': 'number',
+            'format': 'float',
+        },
+        'example': [12.9721, 77.5933],
+        'minItems': 2,
+        'maxItems': 3,
+    }
+
+    COORDINATES_SCHEMA_FOR_LINE_STRING = {
+        'type': 'array',
+        'items': COORDINATES_SCHEMA_FOR_POINT,
+        'example': [[22.4707, 70.0577], [12.9721, 77.5933]],
+        'minItems': 2,
+    }
+
+    GEO_FIELD_TO_SCHEMA = {
+        models.PointField: {
+            'type': {
+                'type': 'string',
+                'enum': ['Point'],
+            },
+            'coordinates': COORDINATES_SCHEMA_FOR_POINT,
+        },
+        models.LineStringField: {
+            'type': {
+                'type': 'string',
+                'enum': ['LineString'],
+            },
+            'coordinates': COORDINATES_SCHEMA_FOR_LINE_STRING,
+        },
+        models.PolygonField: {
+            'type': {
+                'type': 'string',
+                'enum': ['Polygon'],
+            },
+            'coordinates': {
+                'type': 'array',
+                'items': {
+                    **COORDINATES_SCHEMA_FOR_LINE_STRING,
+                    'minItems': 4,
+                },
+                'example': [[0.0, 0.0], [0.0, 50.0], [50.0, 50.0], [50.0, 0.0], [0.0, 0.0]]
+            }
+        }
+    }
+
+    MULTI_FIELD_MAPPING = {
```

Comment:
It looks like `GeometryCollection` isnt being covered by this PR.  It uses `geometries` rather than `geometry`, and I dont see "geometries" mentioned anywhere in this PR.

Reply:
ping @dhaval-mehta . Still no "geometries"

Reply:
@jayvdb man, I am running very busy. I have just merged the latest code in my branch and resolved merge conflicts.
I will sure look into your suggestion but it may take 2-3 months.

Reply:
2-3 months?  Do you mean after this PR is merged?

Unless you have tests passing that include "geometries", it is very likely that merging this PR will cause problems for anyone using `GeometryCollection` in their gis data.

Reply:
> 2-3 months? Do you mean after this PR is merged?
> 
> Unless you have tests passing that include "geometries", it is very likely that merging this PR will cause problems for anyone using `GeometryCollection` in their gis data.

@dhaval-mehta can you clear this concern please?

Reply:
@jayvdb Added support for  GeometryCollectionField and GeometryField